src/Common/Commands.Common/TestMockSupport.cs

```diff
@@ -47,4 +47,37 @@ public static void Delay(TimeSpan duration)
             }
         }
     }
+
+    public class Delay
+    {
+        public const int DefaultMaxDelayInMilliseconds = 60000;
+
+        public const int DefaultBackoffStepInMilliseconds = 5000;
+
+        public const int DefaultDelayInMilliseconds = 0;
+
+        private int backoffStep;
+
+        private int maxDelay;
+
+        private int nextDelay;
+
+        public Delay(int backoffStep = DefaultBackoffStepInMilliseconds, int firstDelay = DefaultDelayInMilliseconds, int maxDelay = DefaultMaxDelayInMilliseconds)
+        {
+            this.backoffStep = backoffStep;
+            this.maxDelay = maxDelay;
+
```

Comment:
Talked with Vivek yesterday. We don't see future reuse of this delay pattern, so I changed it to the simple way to get incremental stepped delay. I committed another change and this class was removed.